Fix inconsistent EXPIRE when ran with flags XX; LT and XX; GT

jainam2907 opened the pull request.

lucifercr07 commented on internal/eval/eval.go:
```diff
@@ -871,7 +871,9 @@ func evaluateAndSetExpiry(subCommands []string, newExpiry uint64, key string,
 	}
 
-	store.SetUnixTimeExpiry(obj, int64(newExpiry))
+	if shouldSetExpiry {
```
Can we please add some unit/integration tests for this scenario?
Also can you confirm the successful run of the tcl test behavior. The steps to run the test cases are mentioned in the `README` of the [dice-tests repository](https://github.com/AshwinKul28/dice-tests).

jainam2907 replied:
Yeah, I am on it. Will update the PR in some time. Thanks!

jainam2907 commented:
RCA: The EXPIRE command correctly returned 0 or 1 based on whether it should set the expiration for a key, but it still unconditionally set the expiration regardless of the return value, which caused the errors.

Resolution: I have fixed this issue by ensuring the expiration is updated conditionally, based on the flags passed and the existing expiration of the key. Additionally, I updated the integration tests to include a call to the TTL command to verify that the expiration is accurate, alongside the response code returned by the previous EXPIRE command.

The attached output of the tcl test confirms the behavior is correct now.
[tcltest-output.log](https://github.com/user-attachments/files/16939902/tcltest-output.log)

lucifercr07 reviewed (approved):
LGTM, approved.

JyotinderSingh commented:
Thanks for the fixes @jainam2907! Thanks for the reviews @lucifercr07.
